apps/microsoft-teams/frontend/src/components/config/AccessSection/AccessSection.tsx

```diff
@@ -87,93 +88,84 @@ const AccessSection = (props: Props) => {
           }}
           handleDisconnect={async () => {
             onClose(true);
-            setOrgDetails(defaultOrgDetails);
             await instance.logoutPopup({
               postLogoutRedirectUri: '/',
               account: accounts[0],
             });
+            const msAccountInfo: Omit<AppInstallationParameters, 'notifications'> = {
+              tenantId: '',
+              authenticatedUsername: '',
+              ...defaultOrgDetails,
+            };
             dispatch({
-              type: actions.UPDATE_TENANT_ID,
-              payload: '',
+              type: actions.UPDATE_MS_ACCOUNT_INFO,
+              payload: msAccountInfo,
             });
           }}
         />
       );
     });
   };
 
-  const getOrgDetails = async () => {
+  const getOrgDetails = async (account: AccountInfo): Promise<typeof defaultOrgDetails> => {
```

Comment:
I noticed in the previous implementation, inside a `useEffect`, we were checking the validity of `accounts` and `tenantId`.  I'm assuming that check is no longer necessary?

```js
if (accounts.length && parameters.tenantId) {
```

Probably because this would throw an error, if tenantId and account were invalid?  Is that a true statement?

```js
const authResult = await instance.loginPopup(loginRequest);
const { tenantId, account } = authResult;
```